onadata/apps/main/tests/test_form_gallery.py

```diff
@@ -51,7 +53,8 @@ def _publish_transportation_id_string_starts_with_number_form(self):
                                 "transportation",
                                 "transportation.id_starts_with_num.xls")
         count = XForm.objects.count()
-        TestBase._publish_xls_file(self, xls_path)
+        response = TestBase._publish_xls_file(self, xls_path)
 
-        self.assertEqual(XForm.objects.count(), count + 1)
+        self.assertContains(response, 'Names must begin with a letter')
+        self.assertEqual(XForm.objects.count(), count)
         self.xform = XForm.objects.all().reverse()[0]
```

Comment:
as @magicznyleszek would say, "BAD CODE™"

Reply:
to elaborate, changing these attributes (like `self.xform`) on the fly, in individual tests instead of the class constructor, is probably confusing. (not that it's even part of this PR :wink:)